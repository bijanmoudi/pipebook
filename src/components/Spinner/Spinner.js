import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './index.scss';

const Spinner = ({ size, text, silent, style, className, ...restProps }) => {
	return (
		<div
			className={classnames('spinner', { [className]: className })}
			style={{ ...style, ...(size ? { fontSize: size * 10 } : {}) }}
			{...restProps}
		>
			<span
				{...(silent ? {} : { role: 'alert' })}
				aria-busy="true"
				className="visuallyhidden"
			>
				{text}
			</span>
		</div>
	);
};

Spinner.propTypes = {
	size: PropTypes.number,
	text: PropTypes.string,
	silent: PropTypes.bool
};

Spinner.defaultProps = {
	size: null,
	text: 'Loading ...',
	silent: false
};

export default Spinner;
