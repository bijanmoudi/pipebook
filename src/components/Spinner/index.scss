/*======================================
=            Spinner Styles            =
======================================*/

@import '../../assets/styles/helpers/colors';
@import '../../assets/styles/helpers/units';

@keyframes spin {
	0% {
		transform: rotate(0);
	}
	to {
		transform: rotate(360deg);
	}
}

.spinner {
	display: inline-block;
	vertical-align: middle;
	width: em(24);
	min-width: em(24);
	height: em(24);
	border-radius: 100%;
	border: em(2) solid get-color('gray-light', 0.5);
	border-top-color: get-color('green');
	border-right-color: get-color('green');
	animation: spin 500ms linear infinite;
}
