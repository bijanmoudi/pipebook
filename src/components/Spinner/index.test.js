import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from './Spinner';

const props = {
	size: 5,
	text: '',
	silent: true
};

describe('Testing "Spinner" component', () => {
	it('Rendering a loading spinner without text', () => {
		const tree = renderer.create(<Spinner text={props.text} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering a big silent spinner with default text', () => {
		const tree = renderer
			.create(<Spinner size={props.size} silent={props.silent} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
