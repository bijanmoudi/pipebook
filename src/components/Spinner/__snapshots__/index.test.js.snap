// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Spinner" component Rendering a big silent spinner with default text 1`] = `
<div
  className="spinner"
  style={
    Object {
      "fontSize": 50,
    }
  }
>
  <span
    aria-busy="true"
    className="visuallyhidden"
  >
    Loading ...
  </span>
</div>
`;

exports[`Testing "Spinner" component Rendering a loading spinner without text 1`] = `
<div
  className="spinner"
  style={Object {}}
>
  <span
    aria-busy="true"
    className="visuallyhidden"
    role="alert"
  >
    
  </span>
</div>
`;
